src/Microsoft.DotNet.Interactive.Jupyter/ZMQ/JupyterConnection.cs

```diff
@@ -0,0 +1,149 @@
+                Key = Guid.NewGuid().ToString(),
+                Transport = "tcp",
+                SignatureScheme = "hmac-sha256"
+            };
+
+            var fileName = $"kernel-{Guid.NewGuid().ToString()}.json";
+            var connectionInfoTempFilePath = Path.Combine(Path.GetTempPath(), fileName);
+            using (var writer = new FileStream(connectionInfoTempFilePath, FileMode.CreateNew))
+            {
+                await JsonSerializer.SerializeAsync(writer, connectionInfo, connectionInfo.GetType(), JsonFormatter.SerializerOptions);
+            }
+
+            var directory = JupyterCommonDirectories.GetRuntimeDirectory();
+            if (!directory.Exists)
+            {
+                directory.Create();
+            }
+            var runtimeConnectionFile = Path.Combine(directory.FullName, fileName);
+            File.Copy(connectionInfoTempFilePath, runtimeConnectionFile);
+            File.Delete(connectionInfoTempFilePath);
+
+            connectionFilePath = runtimeConnectionFile;
+        }
+
+        var kernelProcess = CreateKernelProcess(spec, connectionFilePath);
+        if (kernelProcess == null)
+        {
+            throw new KernelStartException(kernelSpecName, "count not create process.");
+        }
+
+        await Task.Yield();
+
+        if (connectionInfo == null || kernelProcess.HasExited)
+        {
+            throw new KernelStartException(kernelSpecName, $"Process Exited with exit code {kernelProcess.ExitCode}. Ensure you are running in the correct environment.");
+        }
+
+        var kernelConnection = new ZMQKernelConnection(connectionInfo, kernelProcess, kernelSpecName);
+        return kernelConnection;
+    }
+
+    private Process CreateKernelProcess(KernelSpec spec, string connectionFilePath)
+    {
+        List<string> kernelArgs = new(spec?.CommandArguments);
+
+        if (kernelArgs.Count == 0)
+        {
+            return null;
+        }
+
+        // point to the connection file with port information
+        var indexOfConnectionFile = kernelArgs.FindIndex(s => s == "{connection_file}");
+        kernelArgs[indexOfConnectionFile] = $"\"{connectionFilePath}\"";
+
+        // use the process info in kernel spec and replace the {connection_file} in the data with file path
+        // spawn a child process. This will create ZMQ sockets on the kernel. 
+        var command = kernelArgs[0];
+        var arguments = string.Join(" ", kernelArgs.Skip(1));
+
+        Logger kernelLog = new(spec.Name);
+        var kernelProcess = CommandLine.StartProcess(command,
+                                                     arguments,
+                                                     null,
+                                                     o => kernelLog.Info(o),
+                                                     err => kernelLog.Error(err));
+        return kernelProcess;
+    }
+
+    public void Dispose()
```

Comment:
If the `Dispose` isn't always needed, then having `IJupyterConnection` implement `IDisposable` is probably not needed. Implementations can implement it (or `IAsyncDisposable`) on an as-needed basis.

Reply:
will do in a follow up PR.